In [235]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import os
from collections import Counter
print(os.listdir("../Downloads"))

['.ipynb_checkpoints', '1 - Зависимости.ppt', '1М_1Р_1М.28-29.09.2019. И. Михнева.zip', '2М_1Р_2М.12-13.10.2019.И. Михнева.zip', '5853585.a4.zip', 'Anaconda3-2019.10-Windows-x86_64.exe', 'data.csv', 'data_sf.xls', 'desktop.ini', 'Dilts.zip', 'download (1).png', 'download (10).png', 'download (11).png', 'download (12).png', 'download (13).png', 'download (14).png', 'download (2).png', 'download (3).png', 'download (4).png', 'download (5).png', 'download (6).png', 'download (7).png', 'download (8).png', 'download (9).png', 'download(0).png', 'Git-2.25.0-64-bit.exe', 'GitHubDesktopSetup.exe', 'happy bd.jpg', 'IMG_20190928_175811_resized_20190928_065834078.jpg', 'pycharm-community-2019.3.1 (1).exe', 'pycharm-community-2019.3.1 (2).exe', 'pycharm-community-2019.3.1.exe', 'pycharm-community-anaconda-2019.3.1.exe', 'Python A1-1 (15.12.2019).ipynb', 'Python A4 (12.05.2019).ipynb', 'python-3.8.1.exe', 'Skype-8.53.0.85 (1).exe', 'Skype-8.53.0.85.exe', 'SlackSetup.exe', 'Statistika_i_kotiki_Savel

In [263]:
data = pd.read_csv('../Downloads/data.csv')
data.tail(10)

imdb_id  popularity    budget    revenue  \
1880  tt0180073    0.309754  13500000    7060876   
1881  tt0190374    0.284476  35000000   15185241   
1882  tt0213790    0.279624  24000000   13700000   
1883  tt0217756    0.196668  24000000   12372410   
1884  tt0122459    0.249219  24000000   36609995   
1885  tt0120903    0.182246  75000000  157299717   
1886  tt0192255    0.123920  22000000   13555988   
1887  tt0131704    0.104908  76000000   35134820   
1888  tt0162983    0.031018  40000000   36037909   
1889  tt0163676    0.015997  15000000    5217498   

                            original_title  \
1880                                Quills   
1881                    Dungeons & Dragons   
1882                        The Ladies Man   
1883                       Ready to Rumble   
1884                          Return to Me   
1885                                 X-Men   
1886                    The Little Vampire   
1887  The Adventures of Rocky & Bullwinkle   
1888                            Hanging Up   
1889                          The In Crowd   

                                                   cast          director  \
1880  Geoffrey Rush|Kate Winslet|Joaquin Phoenix|Mic...    Philip Kaufman   
1881  Justin Whalin|Jeremy Irons|Thora Birch|Bruce P...  Courtney Solomon   
1882  Tim Meadows|Karyn Parsons|Billy Dee Williams|J...   Reginald Hudlin   
1883  David Arquette|Scott Caan|Oliver Platt|Rose Mc...     Brian Robbins   
1884  David Duchovny|Minnie Driver|Carroll O'Connor|...       Bonnie Hunt   
1885  Patrick Stewart|Hugh Jackman|Ian McKellen|Hall...      Bryan Singer   
1886  Richard E. Grant|Jonathan Lipnicki|Jim Carter|...          Uli Edel   
1887  Rene Russo|Jason Alexander|Piper Perabo|Randy ...       Des McAnuff   
1888  Meg Ryan|Diane Keaton|Lisa Kudrow|Walter Matth...      Diane Keaton   
1889  Susan Ward|Lori Heuring|Matthew Settle|Nathan ...      Mary Lambert   

                                                tagline  \
1880          There are no bad words... only bad deeds.   
1881                                   This is no game.   
1882        He's cool. He's clean. He's a love machine.   
1883       They're headed to the big time...face first.   
1884                   A comedy straight from the heart   
1885                                   Evolution Begins   
1886  They're not just best friends, they're blood b...   
1887            This summer it's not the same old bull.   
1888                   Every family has a few hang-ups.   
1889                       What would you do to get in?   

                                               overview  runtime  \
1880  A nobleman with a literary flair, the Marquis ...      124   
1881  The classic role-playing game comes to life in...      107   
1882  Because of his salacious language, late-night ...       84   
1883  Two slacker wrestling fans are devastated by t...      107   
1884  It took a lot of cajoling to get Bob (Duchovny...      115   
1885  Two mutants, Rogue and Wolverine, come to a pr...      104   
1886  Based on the popular books, the story tells of...       95   
1887  Rocky and Bullwinkle have been living off the ...       88   
1888  A trio of sisters bond over their ambivalence ...       94   
1889  A mentally disturbed young woman takes a job a...      105   

                                        genres  \
1880                                     Drama   
1881  Action|Adventure|Fantasy|Science Fiction   
1882                                    Comedy   
1883                       Action|Comedy|Drama   
1884                      Romance|Comedy|Drama   
1885          Adventure|Action|Science Fiction   
1886                     Horror|Family|Foreign   
1887  Adventure|Animation|Action|Comedy|Family   
1888                              Comedy|Drama   
1889                                  Thriller   

                                   production_companies release_date  \
1880  Fox Searchlight Pictures|Industry Entertainmen...   

In [237]:
display(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1890 entries, 0 to 1889
Data columns (total 16 columns):
imdb_id                 1890 non-null object
popularity              1890 non-null float64
budget                  1890 non-null int64
revenue                 1890 non-null int64
original_title          1890 non-null object
cast                    1890 non-null object
director                1890 non-null object
tagline                 1890 non-null object
overview                1890 non-null object
runtime                 1890 non-null int64
genres                  1890 non-null object
production_companies    1890 non-null object
release_date            1890 non-null object
vote_count              1890 non-null int64
vote_average            1890 non-null float64
release_year            1890 non-null int64
dtypes: float64(2), int64(5), object(9)
memory usage: 236.4+ KB


None

In [352]:
len(data)

1890

# Предобработка датасета

In [353]:
answer_ls = [] # создадим список с ответами. сюда будем добавлять ответы по мере прохождения теста
# сюда можем вписать создание новых колонок в датасете

In [26]:
data_id_budget = data['budget']
i_max = data_id_budget.idxmax()
data.iloc[i_max]['imdb_id']
# the right answer is 4

'tt1032751'

# 1. У какого фильма из списка самый большой бюджет?
Варианты ответов:
1. The Dark Knight Rises (tt1345836)
2. Spider-Man 3 (tt0413300)
3. Avengers: Age of Ultron (tt2395427)
4. The Warrior's Way	(tt1032751)
5. Pirates of the Caribbean: On Stranger Tides (tt1298650)

In [354]:
# тут вводим ваш ответ и добавлем в его список ответов (сейчас для примера стоит "1")
answer_ls.append(4)

In [355]:
answer_ls[0]

4

In [37]:
data_time = data['runtime']
i_time_max = data_time.idxmax()
data.iloc[i_time_max]['imdb_id']
#  the right answer is 2

'tt0279111'

# 2. Какой из фильмов самый длительный (в минутах)
1. The Lord of the Rings: The Return of the King	(tt0167260)
2. Gods and Generals	(tt0279111)
3. King Kong	(tt0360717)
4. Pearl Harbor	(tt0213149)
5. Alexander	(tt0346491)

In [356]:
answer_ls.append(2)

In [357]:
answer_ls[1]

2

In [41]:
data_time = data['runtime']
i_time_min = data_time.idxmin()
data.iloc[i_time_min]['imdb_id']
#  the right answer is 3

'tt1449283'

# 3. Какой из фильмов самый короткий (в минутах)
Варианты ответов:

1. Home on the Range	tt0299172
2. The Jungle Book 2	tt0283426
3. Winnie the Pooh	tt1449283
4. Corpse Bride	tt0121164
5. Hoodwinked!	tt0443536

In [358]:
answer_ls.append(3)

In [359]:
answer_ls

[4, 2, 3]

In [45]:
data['runtime'].mean()
#  the right answer is 2

109.65343915343915

# 4. Средняя длительность фильма?

Варианты ответов:
1. 115
2. 110
3. 105
4. 120
5. 100


In [360]:
answer_ls.append(2)

In [361]:
answer_ls

[4, 2, 3, 2]

In [48]:
data['runtime'].median()
#  the right answer is 1

106.5

# 5. Средняя длительность фильма по медиане?
Варианты ответов:
1. 106
2. 112
3. 101
4. 120
5. 115




In [362]:
answer_ls.append(1)

In [363]:
answer_ls

[4, 2, 3, 2, 1]

In [51]:
data_net_income = data['revenue'] - data['budget']
i_income_max = data_net_income.idxmax()
data.iloc[i_income_max]['imdb_id']
#  the right answer is 5

'tt0499549'

# 6. Какой самый прибыльный фильм?
Варианты ответов:
1. The Avengers	tt0848228
2. Minions	tt2293640
3. Star Wars: The Force Awakens	tt2488496
4. Furious 7	tt2820852
5. Avatar	tt0499549

In [364]:
answer_ls.append(5)

In [365]:
answer_ls

[4, 2, 3, 2, 1, 5]

In [57]:
data_net_income = data['revenue'] - data['budget']
i_income_min = data_net_income.idxmin()
data.iloc[i_income_min]['imdb_id']
#  the right answer is 2

'tt1032751'

# 7. Какой фильм самый убыточный?
Варианты ответов:
1. Supernova tt0134983
2. The Warrior's Way tt1032751
3. Flushed Away	tt0424095
4. The Adventures of Pluto Nash	tt0180052
5. The Lone Ranger	tt1210819

In [366]:
answer_ls.append(2)

In [367]:
answer_ls

[4, 2, 3, 2, 1, 5, 2]

In [60]:
len(data[data.revenue >= data.budget])
#  the right answer is 1

1478

# 8. Сколько всего фильмов в прибыли?
Варианты ответов:
1. 1478
2. 1520
3. 1241
4. 1135
5. 1398


In [368]:
answer_ls.append(1)

In [369]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1]

In [66]:
data_2008 = data[data['release_year'] == 2008]
data_2008_net_income = data_2008['revenue'] - data_2008['budget']
i_income_max = data_2008_net_income.idxmax()
data.iloc[i_income_max]['imdb_id']
#  the right answer is 4

'tt0468569'

# 9. Самый прибыльный фильм в 2008 году?
Варианты ответов:
1. Madagascar: Escape 2 Africa	tt0479952
2. Iron Man	tt0371746
3. Kung Fu Panda	tt0441773
4. The Dark Knight	tt0468569
5. Mamma Mia!	tt0795421

In [370]:
answer_ls.append(4)

In [371]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4]

In [76]:
data_2012_2014 = data[(data['release_year'] >= 2012)& (data['release_year'] <= 2014)]
data_2012_2014_net_income = data_2012_2014['revenue'] - data_2012_2014['budget']
i_income_min = data_2012_2014_net_income.idxmin()
data.iloc[i_income_min]['imdb_id']
#  the right answer is 5

'tt1210819'

# 10. Самый убыточный фильм за период с 2012 по 2014 (включительно)?
Варианты ответов:
1. Winter's Tale	tt1837709
2. Stolen	tt1656186
3. Broken City	tt1235522
4. Upside Down	tt1374992
5. The Lone Ranger	tt1210819


In [372]:
answer_ls.append(5)

In [373]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5]

In [147]:
Counter(data.genres.str.split('|').sum()).most_common()
#  the right answer is 3

[('Drama', 782),
 ('Comedy', 683),
 ('Thriller', 597),
 ('Action', 583),
 ('Adventure', 416),
 ('Crime', 315),
 ('Romance', 308),
 ('Family', 260),
 ('Science Fiction', 248),
 ('Fantasy', 223),
 ('Horror', 176),
 ('Mystery', 168),
 ('Animation', 139),
 ('Music', 64),
 ('History', 62),
 ('War', 58),
 ('Western', 20),
 ('Documentary', 8),
 ('Foreign', 2)]

# 11. Какого жанра фильмов больше всего?
Варианты ответов:
1. Action
2. Adventure
3. Drama
4. Comedy
5. Thriller

In [374]:
answer_ls.append(3)

In [375]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3]

In [148]:
data.profit = data.revenue - data.budget
profitfilms = data[data.profit > 0]
Counter(profitfilms.genres.str.split('|').sum()).most_common()
#  the right answer is 1

[('Drama', 560),
 ('Comedy', 551),
 ('Thriller', 446),
 ('Action', 444),
 ('Adventure', 337),
 ('Romance', 242),
 ('Crime', 231),
 ('Family', 226),
 ('Science Fiction', 195),
 ('Fantasy', 188),
 ('Horror', 150),
 ('Animation', 120),
 ('Mystery', 119),
 ('Music', 47),
 ('History', 46),
 ('War', 41),
 ('Western', 12),
 ('Documentary', 7)]

# 12. Какого жанра среди прибыльных фильмов больше всего?
Варианты ответов:
1. Drama
2. Comedy
3. Action
4. Thriller
5. Adventure

In [376]:
answer_ls.append(1)

In [378]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1]

In [150]:
Counter(data.director).most_common()
#  the right answer is 3

[('Steven Soderbergh', 13),
 ('Ridley Scott', 12),
 ('Clint Eastwood', 12),
 ('Steven Spielberg', 10),
 ('Shawn Levy', 10),
 ('Antoine Fuqua', 8),
 ('Ron Howard', 8),
 ('M. Night Shyamalan', 8),
 ('Christopher Nolan', 8),
 ('Peter Jackson', 8),
 ('Michael Bay', 8),
 ('Bobby Farrelly|Peter Farrelly', 8),
 ('Tim Burton', 8),
 ('Todd Phillips', 8),
 ('Adam Shankman', 8),
 ('Gore Verbinski', 8),
 ('Brett Ratner', 7),
 ('Paul W.S. Anderson', 7),
 ('Lasse Hallström', 7),
 ('Robert Rodriguez', 7),
 ('Dennis Dugan', 7),
 ('Marc Forster', 7),
 ('Francis Lawrence', 6),
 ('Adam McKay', 6),
 ('Robert Zemeckis', 6),
 ('Andy Fickman', 6),
 ('David Fincher', 6),
 ('Tim Story', 6),
 ('Zack Snyder', 6),
 ('Robert Luketic', 6),
 ('Sam Raimi', 6),
 ('Tony Scott', 6),
 ('Martin Scorsese', 6),
 ('Brian Robbins', 6),
 ('Rob Cohen', 6),
 ('Peter Berg', 6),
 ('Peter Segal', 6),
 ('Raja Gosnell', 6),
 ('J.J. Abrams', 5),
 ('Sam Mendes', 5),
 ('Chris Columbus', 5),
 ('Quentin Tarantino', 5),
 ('Peyton Reed', 5

# 13. Кто из режиссеров снял больше всего фильмов?
Варианты ответов:
1. Steven Spielberg
2. Ridley Scott 
3. Steven Soderbergh
4. Christopher Nolan
5. Clint Eastwood

In [379]:
answer_ls.append(3)

In [380]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3]

In [109]:
data_profitable_dir = data[(data.revenue > data.budget)]['director']
data_profitable_dir.describe(include = 'object')
#  the right answer is 4

count             1478
unique             774
top       Ridley Scott
freq                12
Name: director, dtype: object

# 14. Кто из режиссеров снял больше всего Прибыльных фильмов?
Варианты ответов:
1. Steven Soderbergh
2. Clint Eastwood
3. Steven Spielberg
4. Ridley Scott
5. Christopher Nolan

In [381]:
answer_ls.append(4)

In [382]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4]

In [121]:
profit = data[['budget','revenue','director']]
(profit.groupby(['director']).sum()['revenue'] - profit.groupby(['director']).sum()['budget']).idxmax()
#  the right answer is 5

'Peter Jackson'

# 15. Кто из режиссеров принес больше всего прибыли?
Варианты ответов:
1. Steven Spielberg
2. Christopher Nolan
3. David Yates
4. James Cameron
5. Peter Jackson


In [383]:
answer_ls.append(5)

In [384]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5]

In [386]:
actors_list = ['Emma Watson', 'Johnny Depp', 'Michelle Rodriguez', 'Orlando Bloom', 'Rupert Grint']
data.profit = data.revenue - data.budget
profit_list =[]
for actor in actors_list:
    actor_profit = 0
    for i in range(len(data)):
        if actor in data.cast.str.split('|')[i]:
            actor_profit += data.profit[i]
    profit_list.append(actor_profit)
# profit_list.index(max(profit_list))
#  the right answer is 1
profit_list

[6666245597, 4776417000, 4608031235, 5148578162, 6408638290]

# 16. Какой актер принес больше всего прибыли?
Варианты ответов:
1. Emma Watson
2. Johnny Depp
3. Michelle Rodriguez
4. Orlando Bloom
5. Rupert Grint

In [387]:
answer_ls.append(1)

In [389]:
len(answer_ls)

16

In [228]:
actors_list = ['Nicolas Cage', 'Danny Huston', 'Kirsten Dunst', 'Jim Sturgess', 'Sami Gayle']

data.profit = data.revenue - data.budget
#for i in range(len(data)):
#    if data.profit[i] < 0:
#        data.profit[i] = 0

profit_list =[]
for actor in actors_list:
    actor_profit = 0
    for i in range(len(data)):
        if (actor in data.cast.str.split('|')[i]) and (data.release_year[i] == 2012):
            actor_profit += data.profit[i]
    profit_list.append(actor_profit)
profit_list.index(min(profit_list))
profit_list
#  the right answer is 3

[-32893443, -32893443, -68109207, -23410657, -32893443]

# 17. Какой актер принес меньше всего прибыли в 2012 году?
Варианты ответов:
1. Nicolas Cage
2. Danny Huston
3. Kirsten Dunst
4. Jim Sturgess
5. Sami Gayle

In [390]:
answer_ls.append(3)

In [391]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3]

In [171]:
films_high_budget = data[data.budget > data.budget.mean()]
Counter(films_high_budget.cast.str.split('|').sum()).most_common()
#  the right answer is 3

[('Matt Damon', 18),
 ('Adam Sandler', 17),
 ('Angelina Jolie', 16),
 ('Tom Cruise', 15),
 ('Samuel L. Jackson', 15),
 ('Eddie Murphy', 15),
 ('Mark Wahlberg', 14),
 ('Hugh Jackman', 14),
 ('Ben Stiller', 14),
 ('Jamie Foxx', 14),
 ('Russell Crowe', 14),
 ('Johnny Depp', 13),
 ('Jude Law', 13),
 ('Bruce Willis', 13),
 ('Will Smith', 13),
 ('Dwayne Johnson', 12),
 ('Ian McKellen', 12),
 ('Brad Pitt', 12),
 ('Owen Wilson', 12),
 ('Cameron Diaz', 12),
 ('Nicolas Cage', 12),
 ('Robert De Niro', 12),
 ('Leonardo DiCaprio', 11),
 ('Daniel Craig', 11),
 ('Robert Downey Jr.', 11),
 ('Steve Buscemi', 11),
 ('Vin Diesel', 10),
 ('Mark Ruffalo', 10),
 ('Jack Black', 10),
 ('Denzel Washington', 10),
 ('Emma Watson', 10),
 ('Gary Oldman', 10),
 ('Tom Hanks', 10),
 ('Orlando Bloom', 10),
 ('Ralph Fiennes', 9),
 ('Cate Blanchett', 9),
 ('Michael Caine', 9),
 ('Ewan McGregor', 9),
 ('Paul Bettany', 9),
 ('Brendan Gleeson', 9),
 ('Shia LaBeouf', 9),
 ('Nicole Kidman', 9),
 ('Colin Farrell', 9),
 ('Rach

# 18. Какой актер снялся в большем количестве высокобюджетных фильмов? (в фильмах где бюджет выше среднего по данной выборке)
Варианты ответов:
1. Tom Cruise
2. Mark Wahlberg 
3. Matt Damon
4. Angelina Jolie
5. Adam Sandler

In [392]:
answer_ls.append(3)

In [393]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3]

In [176]:
genres_list = ['Drama', 'Action', 'Thriller', 'Adventure', 'Crime']
Nick = 'Nicolas Cage'
nicks_films = data[data.cast.str.contains(Nick)]
Counter(nicks_films.genres.str.split('|').sum()).most_common()
#  the right answer is 2

[('Action', 17),
 ('Thriller', 15),
 ('Drama', 12),
 ('Crime', 10),
 ('Fantasy', 8),
 ('Adventure', 7),
 ('Comedy', 6),
 ('Science Fiction', 4),
 ('Mystery', 3),
 ('Animation', 3),
 ('Family', 3),
 ('History', 2),
 ('War', 1),
 ('Horror', 1),
 ('Romance', 1)]

# 19. В фильмах какого жанра больше всего снимался Nicolas Cage?  
Варианты ответа:
1. Drama
2. Action
3. Thriller
4. Adventure
5. Crime

In [394]:
answer_ls.append(2)

In [395]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3, 2]

In [201]:
#studios = {
# 'Universal':['Universal', 'Universal Pictures', 'Universal Studios'],
# 'Paramount':['Paramount Pictures', 'Paramount Pictures Corporation', 'Paramount'],
# 'Columbia':['Columbia Pictures', 'Columbia Pictures Industries, Inc.', 'Columbia'],
# 'Warner Bros':['Warner Bros', 'Warner Bros. Entertainment Inc.', 'WB'],
# '20th Century Fox':['Twentieth Century Fox Film Corporation', '20th Century Fox', '20th Century Fox Film Corporation']}

#for studiotag in studios.keys():
#    for name in studios.values():
#        for i in range(len(data)):
#            data.studio[i] = []
 #           if name in (data.production_companies.str.split('|')[i]):
#                data.studio.append(studiotag)
#data.studio 
uni_data = data[data.production_companies.str.contains('Universal')]
print('Uni =', len(uni_data))
par_data = data[data.production_companies.str.contains('Paramount')]
print('Par =', len(par_data))
col_data = data[data.production_companies.str.contains('Columbia')]
print('Col =', len(col_data))
war_data = data[data.production_companies.str.contains('Warner Bros')|
               data.production_companies.str.contains('WB')]
print('War =', len(war_data))
fox_data = data[data.production_companies.str.contains('Twentieth Century')|
               data.production_companies.str.contains('20th Century')]
print('Fox =', len(fox_data))
#  the right answer is 1

Uni = 185
Par = 134
Col = 172
War = 183
Fox = 135


# 20. Какая студия сняла больше всего фильмов?
Варианты ответа:
1. Universal Pictures (Universal)
2. Paramount Pictures
3. Columbia Pictures
4. Warner Bros
5. Twentieth Century Fox Film Corporation

In [396]:
answer_ls.append(1)

In [397]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3, 2, 1]

In [287]:
data2015 = data[data.release_year == 2015]

uni = len(data2015[data2015.production_companies.str.contains("Universal Pictures")])
print('Uni =', uni)
par = len(data2015[data2015.production_companies.str.contains("Paramount Pictures")])
print('Par =', par)
col = len(data2015[data2015.production_companies.str.contains("Columbia Pictures")])
print('Col =', col)
war = len(data2015[data2015.production_companies.str.contains("Warner Bros")])
print('War =', war)
fox = len(data2015[data2015.production_companies.str.contains("Twentieth Century Fox Film Corporation")])
print('Fox =', fox)
#  the right answer is 4

Uni = 11
Par = 7
Col = 7
War = 12
Fox = 8


# 21. Какая студия сняла больше всего фильмов в 2015 году?
Варианты ответа:
1. Universal Pictures
2. Paramount Pictures
3. Columbia Pictures
4. Warner Bros
5. Twentieth Century Fox Film Corporation

In [398]:
answer_ls.append(4)

In [399]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3, 2, 1, 4]

In [231]:
uni_data = data[data.production_companies.str.contains('Universal')& data.genres.str.contains('Comedy')]
print('Uni =', uni_data.revenue.sum())
par_data = data[data.production_companies.str.contains('Paramount')& data.genres.str.contains('Comedy')]
print('Par =', par_data.revenue.sum())
col_data = data[data.production_companies.str.contains('Columbia')& data.genres.str.contains('Comedy')]
print('Col =', col_data.revenue.sum())
war_data = data[(data.production_companies.str.contains('Warner Bros')|
               data.production_companies.str.contains('WB'))& data.genres.str.contains('Comedy')]
print('War =', war_data.revenue.sum())
dis_data = data[(data.production_companies.str.contains('Walt Disney')|
               data.production_companies.str.contains('Disney'))& data.genres.str.contains('Comedy')]
print('Dis =', dis_data.revenue.sum())
#  the right answer is 2

Uni = 13133749034
Par = 5276561954
Col = 12222735769
War = 7355328916
Dis = 12066316328


# 22. Какая студия заработала больше всего денег в жанре комедий за все время?
Варианты ответа:
1. Warner Bros
2. Universal Pictures (Universal)
3. Columbia Pictures
4. Paramount Pictures
5. Walt Disney

In [400]:
answer_ls.append(2)

In [402]:
len(answer_ls)

22

In [242]:
uni_data = data[data.production_companies.str.contains('Universal')& (data.release_year == 2012)]
print('Uni =', uni_data.revenue.sum())
par_data = data[data.production_companies.str.contains('Paramount')& (data.release_year == 2012)]
print('Par =', par_data.revenue.sum())
col_data = data[data.production_companies.str.contains('Columbia')& (data.release_year == 2012)]
print('Col =', col_data.revenue.sum())
war_data = data[(data.production_companies.str.contains('Warner Bros')|
               data.production_companies.str.contains('WB'))& (data.release_year == 2012)]
print('War =', war_data.revenue.sum())
luc_data = data[(data.production_companies.str.contains('Lucasfilm'))& (data.release_year == 2012)]
print('Luc =', luc_data.revenue.sum())
#  the right answer is 3

Uni = 2901011579
Par = 606726715
Col = 3468406608
War = 1869520056
Luc = 50365377


# 23. Какая студия заработала больше всего денег в 2012 году?
Варианты ответа:
1. Universal Pictures (Universal)
2. Warner Bros
3. Columbia Pictures
4. Paramount Pictures
5. Lucasfilm

In [403]:
answer_ls.append(3)

In [404]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3, 2, 1, 4, 2, 3]

In [220]:
par_data = data[data.production_companies.str.contains('Paramount')]
par_data['profit'] = par_data['revenue'] - par_data['budget']
i_profit_min = par_data.profit.idxmin()
data.iloc[i_profit_min]['imdb_id']
#  the right answer is 1

C:\Users\User\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


'tt0267626'

# 24. Самый убыточный фильм от Paramount Pictures
Варианты ответа:

1. K-19: The Widowmaker tt0267626
2. Next tt0435705
3. Twisted tt0315297
4. The Love Guru tt0811138
5. The Fighter tt0964517

In [405]:
answer_ls.append(1)

In [406]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3, 2, 1, 4, 2, 3, 1]

In [125]:
profit_year = data[['budget','revenue','release_year']]
(profit_year.groupby(['release_year']).sum()['revenue'] - profit_year.groupby(['release_year']).sum()['budget']).idxmax()
#  the right answer is 5


2015

# 25. Какой Самый прибыльный год (заработали больше всего)?
Варианты ответа:
1. 2014
2. 2008
3. 2012
4. 2002
5. 2015

In [407]:
answer_ls.append(5)

In [408]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3, 2, 1, 4, 2, 3, 1, 5]

In [206]:
data_war = data[data.production_companies.str.contains('Warner Bros')|
               data.production_companies.str.contains('WB')]
profit_year = data_war[['budget','revenue','release_year']]
(profit_year.groupby(['release_year']).sum()['revenue'] 
 - profit_year.groupby(['release_year']).sum()['budget']).idxmax()
#  the right answer is 1

2014

# 26. Какой Самый прибыльный год для студии Warner Bros?
Варианты ответа:
1. 2014
2. 2008
3. 2012
4. 2010
5. 2015

In [409]:
answer_ls.append(1)

In [410]:
answer_ls

[4, 2, 3, 2, 1, 5, 2, 1, 4, 5, 3, 1, 3, 4, 5, 1, 3, 3, 2, 1, 4, 2, 3, 1, 5, 1]

In [254]:
data_t = data
data_t['release_date'] = pd.to_datetime(data_t['release_date'], format = '%m/%d/%Y')
# data_t.head()
data_t['month'] = data_t['release_date'].dt.month
# data_t.head()
Counter(data_t.month).most_common()
#  the right answer is 4

[(9, 227),
 (12, 191),
 (10, 186),
 (8, 161),
 (3, 156),
 (4, 149),
 (6, 147),
 (11, 146),
 (7, 142),
 (5, 140),
 (2, 135),
 (1, 110)]

# 27. В каком месяце за все годы суммарно вышло больше всего фильмов?
Варианты ответа:
1. Январь
2. Июнь
3. Декабрь
4. Сентябрь
5. Май

In [411]:
answer_ls.append(4)

In [413]:
len(answer_ls)

27

In [257]:
Counter(data_t.month)[6] + Counter(data_t.month)[7] + Counter(data_t.month)[8]
#  the right answer is 2

450

# 28. Сколько суммарно вышло фильмов летом? (за июнь, июль, август)
Варианты ответа:
1. 345
2. 450
3. 478
4. 523
5. 381

In [414]:
answer_ls.append(2)

In [416]:
len(answer_ls)

28

In [262]:
data_winter = data_t[(data_t.month == 11)|(data_t.month == 12)|(data_t.month == 1)]
#  data_winter.head()
dir_list = ['Steven Soderbergh', 'Christopher Nolan', 'Clint Eastwood', 'Ridley Scott', 'Peter Jackson']
count_list = []
for director in dir_list:
    count_list.append(len(data_winter[data_winter.director.str.contains(director)]))
count_list
#  the right answer is 5

[6, 1, 7, 3, 8]

# 29. Какой режисер выпускает (суммарно по годам) больше всего фильмов зимой?
Варианты ответов:
1. Steven Soderbergh
2. Christopher Nolan
3. Clint Eastwood
4. Ridley Scott
5. Peter Jackson

In [417]:
answer_ls.append(5)

In [419]:
answer_ls[28]

5

In [275]:
best_months_list = []
for year in range(2000,2016):
    month_list = []
    for month in range(1,13):
        month_list.append(data_t[(data_t.release_year == year)&(data_t.month == month)].revenue.sum())
#     print(month_list)
    best_months_list.append(month_list.index(max(month_list)))
Counter(best_months_list).most_common
#  the right answer is 5

<bound method Counter.most_common of Counter({5: 7, 11: 5, 4: 3, 6: 1})>

# 30. Какой месяц чаще всего по годам самый прибыльный?
Варианты ответа:
1. Январь
2. Июнь
3. Декабрь
4. Сентябрь
5. Май

In [420]:
answer_ls.append(5)

In [423]:
len(answer_ls)

30

In [351]:
# data_s = data
# data_s['title_len'] = data_s['original_title'].str.len()
# # data_s['title_len']
# data_s['studio_list'] = data_s.production_companies.str.split('|')
# # data_s['studio_list']
# full_studio_list = []
# for i in range(len(data_s)):
#     for studio in data_s['studio_list'][i]:
#         if studio in full_studio_list:
#             continue
#         else:
#             full_studio_list.append(studio)
# full_studio_list.sort()
# full_studio_list
# title_len_list =[]
# Counter(data_s.production_companies.str.split('|').sum()).most_common()

data_l = data[data.production_companies.str.contains('Universal Pictures')|
             data.production_companies.str.contains('Universal')|
             data.production_companies.str.contains('Warner Bros')|
             data.production_companies.str.contains('Jim Henson Company, The')|
             data.production_companies.str.contains('Paramount Pictures')|
             data.production_companies.str.contains('Four By Two Productions')]
# data_l

print('Uni =', data_l[data_l.production_companies.str.contains('Universal')].title_len.mean())
print('War =', data_l[data_l.production_companies.str.contains('Warner Bros')].title_len.mean())
print('Jim =', data_l[data_l.production_companies.str.contains('Jim Henson Company, The')].title_len.mean())
print('Par =', data_l[data_l.production_companies.str.contains('Paramount Pictures')].title_len.mean())
print('Four =', data_l[data_l.production_companies.str.contains('Four By Two Productions')].title_len.mean())
#  the right answer is 5

Uni = 14.762162162162163
War = 15.251366120218579
Jim = 59.0
Par = 17.0327868852459
Four = 83.0


# 31. Названия фильмов какой студии в среднем самые длинные по количеству символов?
Варианты ответа:
1. Universal Pictures (Universal)
2. Warner Bros
3. Jim Henson Company, The
4. Paramount Pictures
5. Four By Two Productions

In [424]:
answer_ls.append(5)

In [426]:
answer_ls[30]

5

In [302]:
data_n = data[data.production_companies.str.contains('Universal Pictures')|
             data.production_companies.str.contains('Universal')|
             data.production_companies.str.contains('Warner Bros')|
             data.production_companies.str.contains('Jim Henson Company, The')|
             data.production_companies.str.contains('Paramount Pictures')|
             data.production_companies.str.contains('Four By Two Productions')]
# len(data_n)
data_n['n_of_words'] = data_n.original_title.apply(lambda x: len(x.split(' ')))
# data_n.n_of_words
print('Uni =', data_n[data_n.production_companies.str.contains('Universal')].n_of_words.mean())
print('War =', data_n[data_n.production_companies.str.contains('Warner Bros')].n_of_words.mean())
print('Jim =', data_n[data_n.production_companies.str.contains('Jim Henson Company, The')].n_of_words.mean())
print('Par =', data_n[data_n.production_companies.str.contains('Paramount Pictures')].n_of_words.mean())
print('Four =', data_n[data_n.production_companies.str.contains('Four By Two Productions')].n_of_words.mean())
#  the right answer is 5

Uni = 2.627027027027027
War = 2.73224043715847
Jim = 10.0
Par = 2.8688524590163933
Four = 12.0


C:\Users\User\Anaconda3\lib\site-packages\ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


# 32. Названия фильмов какой студии в среднем самые длинные по количеству слов?
Варианты ответа:
1. Universal Pictures (Universal)
2. Warner Bros
3. Jim Henson Company, The
4. Paramount Pictures
5. Four By Two Productions

In [427]:
answer_ls.append(5)

In [429]:
answer_ls[31]

5

In [307]:
data['original_title'] = data.original_title.apply(lambda x:str(x).lower())
len(sorted(Counter(data.original_title.str.split(' ').sum()).most_common()))
#  the right answer is 3

2462

# 33. Сколько разных слов используется в названиях фильмов?(без учета регистра)
Варианты ответа:
1. 6540
2. 1002
3. 2461
4. 28304
5. 3432

In [430]:
answer_ls.append(3)

In [432]:
len(answer_ls)

33

In [311]:
data[data.vote_average >= data.vote_average.max()*0.97]
#  the right answer is 1

imdb_id  popularity     budget     revenue  \
9     tt2096673    6.326804  175000000   853708609   
34    tt3170832    3.557846    6000000    35401758   
118   tt0816692   24.949134  165000000   621752480   
119   tt2015381   14.311205  170000000   773312399   
125   tt2084970    8.110711   14000000   233555708   
128   tt2267998    6.438727   61000000   369330363   
138   tt2278388    4.930820   30000000   174600318   
370   tt1375666    9.363643  160000000   825500000   
600   tt0468569    8.466668  185000000  1001921825   
873   tt0253474    2.364204   35000000   120072577   
1082  tt0167260    7.122455   94000000  1118888979   
1184  tt0993846    4.877927  100000000   392000694   
1192  tt2024544    3.625529   20000000   187000000   
1801  tt0209144    3.095625    9000000    39723096   

                                     original_title  \
9                                        inside out   
34                                             room   
118                                    interstellar   
119                         guardians of the galaxy   
125                              the imitation game   
128                                       gone girl   
138                        the grand budapest hotel   
370                                       inception   
600                                 the dark knight   
873                                     the pianist   
1082  the lord of the rings: the return of the king   
1184                        the wolf of wall street   
1192                               12 years a slave   
1801                                        memento   

                                                   cast           director  \
9     Amy Poehler|Phyllis Smith|Richard Kind|Bill Ha...        Pete Docter   
34    Brie Larson|Jacob Tremblay|Joan Allen|Sean Bri...   Lenny Abrahamson   
118   Matthew McConaughey|Jessica Chastain|Anne Hath...  Christopher Nolan   
119   Chris Pratt|Zoe Saldana|Dave Bautista|Vin Dies...         James Gunn   
125   Benedict Cumberbatch|Keira Knightley|Matthew G...      Morten Tyldum   
128   Ben Affleck|Rosamund Pike|Carrie Coon|Neil Pat...      David Fincher   
138   Ralph Fiennes|Tony Revolori|F. Murray Abraham|...       Wes Anderson   
370   Leonardo DiCaprio|Joseph Gordon-Levitt|Ellen P...  Christopher Nolan   
600   Christian Bale|Michael Caine|Heath Ledger|Aaro...  Christopher Nolan   
873   Adrien Brody|Thomas Kretschmann|Frank Finlay|M...     Roman Polanski   
1082  Elijah Wood|Ian McKellen|Viggo Mortensen|Liv T...      Peter Jackson   
1184  Leonardo DiCaprio|Jonah Hill|Margot Robbie|Kyl...    Martin Scorsese   
1192  Chiwetel Ejiofor|Michael Fassbender|Lupita Nyo...      Steve McQueen   
1801  Guy Pearce|Carrie-Anne Moss|Joe Pantoliano|Mar...  Christopher Nolan   

                                                tagline  \
9              Meet the little voices inside your head.   
34                             Love knows no boundaries   
118   Mankind was born on Earth. It was never meant ...   
119                         All heroes start somewhere.   
125   The true enigma was the man who cracked the code.   
128         You don't know what you've got 'til it's...   
138             A perfect holiday without leaving home.   
370                Your mind is the scene of the crime.   
600                                     Why So Serious?   
873   Music was his passion. Survival was his master...   
1082                    The eye of the enemy is moving.   
1184                                EARN. SPEND. PARTY.   
1192    The extraordinary true story of Solomon Northup   
1801                  Some memories are best forgotten.   

                                               overview  runtime  \
9     Growing up can be a bumpy road, and it's no ex...       94   
34    Jack is a young boy of 5 years old who has liv...      117   
118   Interstellar chronicles the adventures of a gr...      169   
119   Light years from Earth, 26 years aft

# 34. Какие фильмы входят в 1 процент лучших по рейтингу?
Варианты ответа:
1. Inside Out, Gone Girl, 12 Years a Slave
2. BloodRayne, The Adventures of Rocky & Bullwinkle
3. The Lord of the Rings: The Return of the King
4. 300, Lucky Number Slevin

In [433]:
answer_ls.append(1)

In [435]:
answer_ls[33]

1

In [335]:
def number_of_films(pair):
    return len(data[data.cast.str.contains(pair[0]) & data.cast.str.contains(pair[1])])

actors_pairs = [['Johnny Depp', 'Helena Bonham Carter'],
               ['Hugh Jackman', 'Ian McKellen'],
               ['Vin Diesel', 'Paul Walker'],
               ['Adam Sandler', 'Kevin James'],
               ['Daniel Radcliffe', 'Rupert Grint']]
for pair in actors_pairs:
    print(pair, ' ',number_of_films(pair) )
#  the right answer is 5

['Johnny Depp', 'Helena Bonham Carter']   6
['Hugh Jackman', 'Ian McKellen']   5
['Vin Diesel', 'Paul Walker']   5
['Adam Sandler', 'Kevin James']   5
['Daniel Radcliffe', 'Rupert Grint']   8


# 35. Какие актеры чаще всего снимаются в одном фильме вместе
Варианты ответа:
1. Johnny Depp & Helena Bonham Carter
2. Hugh Jackman & Ian McKellen
3. Vin Diesel & Paul Walker
4. Adam Sandler & Kevin James
5. Daniel Radcliffe & Rupert Grint

In [436]:
answer_ls.append(5)

In [438]:
answer_ls[34]

5

In [328]:
def profit_probability(director):
    data_d = data[data.director.str.contains(director)]
    data_d['profit'] = data_d['revenue'] - data_d['budget']
    # len(data_d.profit)
    return len(data_d[data_d.profit > 0])/len(data_d.profit)

dir_list = ['Quentin Tarantino', 'Steven Soderbergh', 'Robert Rodriguez', 'Christopher Nolan', 'Clint Eastwood' ]
for director in dir_list:
    print(director,' ', profit_probability(director))
#  the right answer is 4

Quentin Tarantino   0.8571428571428571
Steven Soderbergh   0.7692307692307693
Robert Rodriguez   0.7272727272727273
Christopher Nolan   1.0
Clint Eastwood   0.8333333333333334


C:\Users\User\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


# 36. У какого из режиссеров выше вероятность выпустить фильм в прибыли? (5 баллов)101
Варианты ответа:
1. Quentin Tarantino
2. Steven Soderbergh
3. Robert Rodriguez
4. Christopher Nolan
5. Clint Eastwood

In [439]:
answer_ls.append(4)

# Submission

In [441]:
len(answer_ls)

36

In [442]:
pd.DataFrame({'Id':range(1,len(answer_ls)+1), 'Answer':answer_ls}, columns=['Id', 'Answer'])

Id  Answer
0    1       4
1    2       2
2    3       3
3    4       2
4    5       1
5    6       5
6    7       2
7    8       1
8    9       4
9   10       5
10  11       3
11  12       1
12  13       3
13  14       4
14  15       5
15  16       1
16  17       3
17  18       3
18  19       2
19  20       1
20  21       4
21  22       2
22  23       3
23  24       1
24  25       5
25  26       1
26  27       4
27  28       2
28  29       5
29  30       5
30  31       5
31  32       5
32  33       3
33  34       1
34  35       5
35  36       4